src/query/expression/src/kernels/group_by_hash.rs

```diff
@@ -781,6 +788,104 @@ pub unsafe fn serialize_column_binary(column: &Column, row: usize, row_space: &m
     }
 }
 
+/// The batch version of [`serialize_columns_no_batch`]. Try to leverage SIMD to improve performance.
+///
+/// To serialize a column in batch, we need to preallocate a deterministic memory space for each row of the target column at once.
+pub fn serialize_columns_in_batch(
+    columns: &[Column],
+    num_rows: usize,
+    max_bytes_per_row: usize,
+) -> StringColumn {
+    let mut data: Vec<u8> = Vec::with_capacity(num_rows * max_bytes_per_row);
```

Comment:
This data should be a filed with type `Vec<Vec<u8>>`, thus we could reuse the data in next batch.